File: iosApp/iosApp/Logger/FirebaseLogger.swift
//
//  FirebaseLogger.swift
//  iosApp
//
//  Created by ABDULKARIM ABDULRAHMAN on 13/07/2025.
//  Copyright © 2025 orgName. All rights reserved.
//

import Swift
import GPT_Investor
import FirebaseAnalytics

class FirebaseLogger: AnalyticsLogger {

    func identifyUser(eventName: String, params: [String: Any]) {
        guard let userId = params["user_id"] as? String else {
            print("User ID not found in params")
            return
        }
        Analytics.setUserID(userId)
        params.forEach { (key, value) in
            if key != "user_id" {
                Analytics.setUserProperty(value as? String, forName: key)
            }
        }
    }

    func logEvent(eventName: String, params: [String: Any]) {
        Analytics.logEvent(eventName, parameters: params)
    }

    func logViewEvent(screenName: String) {

    }

    func resetUser(eventName: String) {
        Analytics.resetAnalyticsData()
    }
}

File: analytics/src/androidMain/kotlin/com/thejawnpaul/gptinvestor/analytics/firebase/FirebaseLogger.kt
package com.thejawnpaul.gptinvestor.analytics.firebase

import android.os.Bundle
import com.google.firebase.Firebase
import com.google.firebase.analytics.analytics
import com.thejawnpaul.gptinvestor.analytics.AnalyticsLogger

actual class FirebaseLogger : AnalyticsLogger {

    private val firebaseAnalytics = Firebase.analytics

    actual override fun logEvent(
        eventName: String,
        params: Map<String, Any>
    ) {
        val bundle = Bundle()
        params.forEach { (key, value) ->
            bundle.putString(key, value.toString())
        }
        firebaseAnalytics.logEvent(eventName.lowercase().replace(" ", "_"), bundle)
    }

    actual override fun logViewEvent(screenName: String) {

    }

    actual override fun identifyUser(eventName: String, params: Map<String, Any>) {
        firebaseAnalytics.setUserId(params["user_id"].toString())
    }

    actual override fun resetUser(eventName: String) {
        firebaseAnalytics.resetAnalyticsData()
    }

}